utils/numbers.ml

```diff
@@ -33,6 +33,39 @@ module Int = struct
     if n < 0 then Set.empty else Set.add n (zero_to_n (n-1))
 end
 
+module Int8 = struct
+  type t = int
+
+  let zero = 0
+  let one = 1
+
+  let of_int_exn i =
+    if i < -(1 lsl 8) || i > ((1 lsl 8) - 1) then
```

Comment:
The range check doesn't look right, isn't `-(1 lsl 7), (1 lsl 7) - 1` the valid range? I'd suggest using hexadecimal literals (same for `Int16`).